### PR TITLE
Modified WebApp counter to behave same as Standard counter.

### PR DESCRIPTION
**Pull request by cijothomas**
(minor changes in logging error messages only)

Also fix:
https://github.com/Microsoft/ApplicationInsights-dotnet-server/issues/237

### CODE REVIEW

File: Src/PerformanceCollector/Shared/Implementation/WebAppPerformanceCollector/WebAppPerformanceCollector.cs [hunk 1]
```diff
@@ -86,28 +86,16 @@ public void RefreshCounters()
...
+                bool useInstancePlaceHolder = false;                
+                var pc = PerformanceCounterUtility.CreateAndValidateCounter(perfCounter, null, null, out useInstancePlaceHolder, out error);
+                
                 if (pc != null)
                 {
                     this.RegisterPerformanceCounter(perfCounter, this.GetCounterReportAsName(perfCounter, reportAs), pc.CategoryName, pc.CounterName, pc.InstanceName, useInstancePlaceHolder, false);
-                }
-                else
-                {
-                    this.RegisterPerformanceCounter(perfCounter, this.GetCounterReportAsName(perfCounter, reportAs), string.Empty, perfCounter, string.Empty, useInstancePlaceHolder, false);
```
Comment (kartang) [inline Src/PerformanceCollector/Shared/Implementation/WebAppPerformanceCollector/WebAppPerformanceCollector.cs]: Why is this condition ignored? I believe we can still register the performance counter for web apps if the validation fails. We might just be missing Instance name and Category name?

Comment (cijothomas) [inline Src/PerformanceCollector/Shared/Implementation/WebAppPerformanceCollector/WebAppPerformanceCollector.cs]: I didn't knew that webapp perf counter works even if validation fails.. But if that's the case, why bother validating? Or should we add a separate validator for webappperf counter?

Comment (kartang) [inline Src/PerformanceCollector/Shared/Implementation/WebAppPerformanceCollector/WebAppPerformanceCollector.cs]: We do the validation to get the category name and instance name. If available, we pass them as parameters, and if not, we use empty strings.

Comment (cijothomas) [inline Src/PerformanceCollector/Shared/Implementation/WebAppPerformanceCollector/WebAppPerformanceCollector.cs]: got it. I will bring back the original behavior and add a comment explaining this.

**Comment by cijothomas** [comment]
@kartang can you take a look now?

**Comment by kartang** [comment]
:shipit:

**Pull request merged**